### ISSUE TITLE
(Potentially) Invalid Project

### ISSUE THREAD

**Issue by WumiHam** [opentoonz/opentoonz#5672]
### Description

Within maybe half an hour working on a scene, OpenToonz crashes. I cannot see any specific action forcing the crash. 

### Steps to Reproduce

1. The last step was to move the Py Light Level (white Trapez).   
2. But previous crashes came after different actions.

### Screenshots, Video & Crash Logs

![image](https://github.com/user-attachments/assets/dbb02927-2909-4e9f-8961-129b59fbd759)

OpenToonz 1.7.1  (Build Mar 14 2023)

Report Date: 2024-12-10 18:03:21
Crash Reason: EXCEPTION_ACCESS_VIOLATION

Build ABI: x86_64-little_endian-llp64
Operating System: Windows 10 Version 2009
OS Kernel: 10.0.22631
CPU Threads: 20

GPU Vendor: Intel
GPU Model: Intel(R) Iris(R) Xe Graphics
GPU Version: 4.6.0 - Build 32.0.101.5972

Crash File: C:\Users\klaus\AppData\Local\OpenToonz\OpenToonz\crash\Crash-20241210-180321.log
Mini Dump File: C:\Users\klaus\AppData\Local\OpenToonz\OpenToonz\crash\Crash-20241210-180321.dmp

Application Dir: C:/Program Files/OpenToonz
Stuff Dir: C:\OpenToonz stuff

Project Name: Convenience
Scene Name: Hippotenuse
Project Path: C:\Users\klaus\OneDrive\Animation\Projects\Convenience\Convenience_otprj.xml
Scene Path: C:\Users\klaus\OneDrive\Animation\Projects\Convenience\scenes\Hippotenuse.tnz

==== Modules ====
C:\Program Files\OpenToonz\OpenToonz.exe
C:\windows\SYSTEM32\ntdll.dll
C:\windows\System32\KERNEL32.DLL
C:\windows\System32\KERNELBASE.dll
C:\windows\System32\USER32.dll
C:\windows\System32\win32u.dll
C:\windows\System32\GDI32.dll
C:\windows\System32\gdi32full.dll
C:\windows\System32\msvcp_win.dll
C:\windows\System32\ucrtbase.dll
C:\Program Files\OpenToonz\Qt5Xml.dll
C:\Program Files\OpenToonz\Qt5PrintSupport.dll
C:\windows\System32\SHELL32.dll
C:\windows\System32\ole32.dll
C:\windows\SYSTEM32\OPENGL32.dll
C:\windows\System32\combase.dll
C:\windows\System32\msvcrt.dll
C:\windows\System32\ADVAPI32.dll
C:\windows\System32\RPCRT4.dll
C:\windows\System32\sechost.dll
C:\windows\System32\OLEAUT32.dll
C:\windows\System32\COMDLG32.dll
C:\windows\System32\bcrypt.dll
C:\windows\System32\shcore.dll
C:\windows\System32\SHLWAPI.dll
C:\Program Files\OpenToonz\Qt5SerialPort.dll
C:\Program Files\OpenToonz\glut64.dll
C:\Program Files\OpenToonz\EDSDK.dll
C:\windows\System32\SETUPAPI.dll
C:\Program Files\OpenToonz\colorfx.dll
C:\windows\System32\WS2_32.dll
C:\Program Files\OpenToonz\image.dll
C:\Program Files\OpenToonz\tnztools.dll
C:\Program Files\OpenToonz\tnzstdfx.dll
C:\Program Files\OpenToonz\tfarm.dll
C:\Program Files\OpenToonz\Qt5Widgets.dll
C:\Program Files\OpenToonz\Qt5Gui.dll
C:\Program Files\OpenToonz\Qt5Core.dll
C:\Program Files\OpenToonz\toonzqt.dll
C:\Program Files\OpenToonz\sound.dll
C:\Program Files\OpenToonz\opencv_world451.dll
C:\Program Files\OpenToonz\toonzlib.dll
C:\Program Files\OpenToonz\Qt5Script.dll
C:\Program Files\OpenToonz\tnzext.dll
C:\Program Files\OpenToonz\tnzbase.dll
C:\Program Files\OpenToonz\tnzcore.dll
C:\Program Files\OpenToonz\Qt5OpenGL.dll
C:\Program Files\OpenToonz\Qt5Multimedia.dll
C:\Program Files\OpenToonz\Qt5Network.dll
C:\windows\SYSTEM32\MPR.dll
C:\windows\SYSTEM32\GLU32.dll
C:\windows\System32\CRYPT32.dll
C:\windows\SYSTEM32\dbghelp.dll
C:\windows\SYSTEM32\WINSPOOL.DRV
C:\Program Files\OpenToonz\MSVCP140.dll
C:\Program Files\OpenToonz\VCRUNTIME140.dll
C:\Program Files\OpenToonz\MSVCP140_1.dll
C:\windows\SYSTEM32\MFPlat.DLL
C:\Program Files\OpenToonz\VCRUNTIME140_1.dll
C:\windows\SYSTEM32\MF.dll
C:\windows\WinSxS\amd64_microsoft.windows.common-controls_6595b64144ccf1df_5.82.22621.3527_none_b43b7f4b638cc64f\COMCTL32.dll
C:\windows\SYSTEM32\WINMM.dll
C:\windows\SYSTEM32\VERSION.dll
C:\windows\SYSTEM32\MSVFW32.dll
C:\windows\SYSTEM32\AVIFIL32.dll
C:\Program Files\OpenToonz\libmypaint-1-4-0.dll
C:\Program Files\OpenToonz\glew32.dll
C:\windows\SYSTEM32\UxTheme.dll
C:\windows\SYSTEM32\dwmapi.dll
C:\windows\SYSTEM32\d3d11.dll
C:\windows\SYSTEM32\dxgi.dll
C:\windows\SYSTEM32\USERENV.dll
C:\windows\SYSTEM32\NETAPI32.dll
C:\Program Files\OpenToonz\Qt5Svg.dll
C:\Program Files\OpenToonz\CONCRT140.dll
C:\windows\SYSTEM32\DNSAPI.dll
C:\windows\SYSTEM32\IPHLPAPI.DLL
C:\windows\SYSTEM32\cfgmgr32.dll
C:\windows\SYSTEM32\MSACM32.dll
C:\Program Files\OpenToonz\libintl-8.dll
C:\Program Files\OpenToonz\libjson-c-2.dll
C:\windows\SYSTEM32\dxcore.dll
C:\Program Files\OpenToonz\libiconv-2.dll
C:\windows\SYSTEM32\dbgcore.DLL
C:\windows\SYSTEM32\NETUTILS.DLL
C:\windows\SYSTEM32\MFCORE.DLL
C:\windows\SYSTEM32\SRVCLI.DLL
C:\windows\SYSTEM32\CRYPTBASE.DLL
C:\windows\System32\IMM32.DLL
C:\windows\System32\NSI.dll
C:\windows\SYSTEM32\kernel.appcore.dll
C:\windows\System32\bcryptPrimitives.dll
C:\windows\SYSTEM32\RTWorkQ.DLL
C:\Program Files\OpenToonz\platforms\qwindows.dll
C:\windows\SYSTEM32\WTSAPI32.dll
C:\windows\SYSTEM32\windows.storage.dll
C:\windows\SYSTEM32\wintypes.dll
C:\windows\SYSTEM32\profapi.dll
C:\windows\SYSTEM32\powrprof.dll
C:\windows\SYSTEM32\UMPDC.dll
C:\Program Files\OpenToonz\styles\qwindowsvistastyle.dll
C:\Program Files\OpenToonz\iconengines\qsvgicon.dll
C:\windows\system32\d3d9.dll
C:\windows\SYSTEM32\directxdatabasehelper.dll
C:\windows\System32\MSCTF.dll
C:\windows\SYSTEM32\textinputframework.dll
C:\windows\SYSTEM32\CoreMessaging.dll
C:\windows\SYSTEM32\CoreUIComponents.dll
C:\windows\System32\clbcatq.dll
C:\Windows\System32\AppXDeploymentClient.dll
C:\Windows\System32\Windows.ApplicationModel.dll
C:\windows\system32\propsys.dll
C:\Windows\System32\Windows.FileExplorer.Common.dll
C:\windows\system32\mssprxy.dll
C:\Windows\System32\Windows.StateRepositoryPS.dll
C:\windows\SYSTEM32\windows.staterepositoryclient.dll
C:\windows\System32\DriverStore\FileRepository\iigd_dch.inf_amd64_4ec1a03daa49235f\igxelpicd64.dll
C:\windows\SYSTEM32\d3d12.dll
C:\windows\System32\DriverStore\FileRepository\iigd_dch.inf_amd64_4ec1a03daa49235f\igdml64.dll
C:\windows\System32\DriverStore\FileRepository\iigd_dch.inf_amd64_4ec1a03daa49235f\igdgmm64.dll
C:\windows\System32\DriverStore\FileRepository\iigd_dch.inf_amd64_4ec1a03daa49235f\igc64.dll
C:\windows\System32\DriverStore\FileRepository\iigd_dch.inf_amd64_4ec1a03daa49235f\igc1464.dll
C:\windows\system32\dataexchange.dll
C:\windows\system32\twinapi.appcore.dll
C:\windows\system32\dwrite.dll
C:\windows\SYSTEM32\WINMMBASE.dll
C:\Program Files\OpenToonz\imageformats\qgif.dll
C:\Program Files\OpenToonz\imageformats\qicns.dll
C:\Program Files\OpenToonz\imageformats\qico.dll
C:\Program Files\OpenToonz\imageformats\qjpeg.dll
C:\Program Files\OpenToonz\imageformats\qsvg.dll
C:\Program Files\OpenToonz\imageformats\qtga.dll
C:\Program Files\OpenToonz\imageformats\qtiff.dll
C:\Program Files\OpenToonz\imageformats\qwbmp.dll
C:\Program Files\OpenToonz\imageformats\qwebp.dll
C:\Program Files\OpenToonz\bearer\qgenericbearer.dll
C:\windows\SYSTEM32\dhcpcsvc6.DLL
C:\windows\SYSTEM32\dhcpcsvc.DLL
C:\windows\System32\mswsock.dll
C:\Windows\System32\rasadhlp.dll
C:\windows\System32\netprofm.dll
C:\windows\System32\npmproxy.dll
C:\windows\system32\napinsp.dll
C:\windows\system32\pnrpnsp.dll
C:\windows\System32\winrnr.dll
C:\windows\system32\wshbth.dll
C:\windows\system32\nlansp_c.dll
C:\windows\System32\fwpuclnt.dll
C:\Program Files\OpenToonz\audio\qtaudio_windows.dll
C:\windows\SYSTEM32\MMDevAPI.DLL
C:\windows\SYSTEM32\wdmaud.drv
C:\windows\SYSTEM32\ksuser.dll
C:\windows\SYSTEM32\AVRT.dll
C:\windows\SYSTEM32\DEVOBJ.dll
C:\windows\SYSTEM32\AUDIOSES.DLL
C:\windows\SYSTEM32\msacm32.drv
C:\windows\SYSTEM32\midimap.dll
C:\windows\SYSTEM32\LINKINFO.dll
C:\windows\system32\winhttp.dll
C:\windows\SYSTEM32\SspiCli.dll
C:\windows\WinSxS\amd64_microsoft.windows.common-controls_6595b64144ccf1df_6.0.22621.4391_none_2715d37f73803e96\comctl32.dll
C:\windows\SYSTEM32\WindowsCodecs.dll
C:\windows\SYSTEM32\TextShaping.dll
C:\windows\SYSTEM32\urlmon.dll
C:\windows\SYSTEM32\iertutil.dll
C:\windows\SYSTEM32\edputil.dll
C:\Windows\System32\thumbcache.dll
==== End ====

==== Backtrace ====
  1> RtlGetLengthWithoutLastFullDosOrNtPathElement [0x7ffe88d87faa] <ntdll.dll>
  2> RtlFindCharInUnicodeString [0x7ffe88d2e662] <ntdll.dll>
  3> KiUserExceptionDispatcher [0x7ffe88db426e] <ntdll.dll>
  4> DumpRegistryKeyDefinitions [0x7ffe3efc8649] <igxelpicd64.dll>
  5> DumpRegistryKeyDefinitions [0x7ffe3ee6cf27] <igxelpicd64.dll>
  6> DumpRegistryKeyDefinitions [0x7ffe3ee6d6c3] <igxelpicd64.dll>
  7> DumpRegistryKeyDefinitions [0x7ffe3ee6b651] <igxelpicd64.dll>
  8> DumpRegistryKeyDefinitions [0x7ffe3ec1f905] <igxelpicd64.dll>
  9> RegisterProcTableCallback [0x7ffe3e4c7f9e] <igxelpicd64.dll>
 10> DumpRegistryKeyDefinitions [0x7ffe3ec11afd] <igxelpicd64.dll>
 11> mergePalette_Overlap [0x7ffe330ca09c] <tnzcore.dll>
 12> tglDraw [0x7ffe330b27ba] <tnzcore.dll>
 13> OutlineStrokeProp::~OutlineStrokeProp [0x7ffe330b1b38] <tnzcore.dll>
 14> tglDraw [0x7ffe330b2ff7] <tnzcore.dll>
 15> Stage::RasterPainter::onVectorImage [0x7ffe35cd4a3e] <toonzlib.dll>
 16> Stage::RasterPainter::onImage [0x7ffe35cd237f] <toonzlib.dll>
 17> Stage::visit [0x7ffe35ccb0cb] <toonzlib.dll>
 18> Stage::visit [0x7ffe35cca980] <toonzlib.dll>
 19> TSmartPointerT<TXshSoundTextLevel>::~TSmartPointerT<TXshSoundTextLevel> [0x7ff6881ffe8a] <OpenToonz.exe>
 20> TSmartPointerT<TXshSoundTextLevel>::~TSmartPointerT<TXshSoundTextLevel> [0x7ff6881fdda4] <OpenToonz.exe>
 21> TSmartPointerT<TXshSoundTextLevel>::~TSmartPointerT<TXshSoundTextLevel> [0x7ff688204535] <OpenToonz.exe>
 22> TSmartPointerT<TXshSoundTextLevel>::~TSmartPointerT<TXshSoundTextLevel> [0x7ff688204914] <OpenToonz.exe>
 23> TTool::pick [0x7ffe3a0d71f4] <tnztools.dll>
 24> PinchTool::getToolType [0x7ffe3a0f5766] <tnztools.dll>
 25> TSmartPointerT<TXshSoundTextLevel>::~TSmartPointerT<TXshSoundTextLevel> [0x7ff68820b52b] <OpenToonz.exe>
 26> TSmartPointerT<TXshSoundTextLevel>::~TSmartPointerT<TXshSoundTextLevel> [0x7ff68820a5f8] <OpenToonz.exe>
 27> QWidget::event [0x7ffe35478066] <Qt5Widgets.dll>
 28> TSmartPointerT<TXshSoundTextLevel>::~TSmartPointerT<TXshSoundTextLevel> [0x7ff68820942b] <OpenToonz.exe>
 29> QApplicationPrivate::notify_helper [0x7ffe354555c0] <Qt5Widgets.dll>
 30> QApplication::notify [0x7ffe354534e0] <Qt5Widgets.dll>
 31> QCoreApplication::notifyInternal2 [0x7ffe334c318a] <Qt5Core.dll>
 32> QApplicationPrivate::sendMouseEvent [0x7ffe354567c5] <Qt5Widgets.dll>
 33> QDesktopWidget::workAreaResized [0x7ffe354a09d2] <Qt5Widgets.dll>
 34> QDesktopWidget::workAreaResized [0x7ffe3549e96e] <Qt5Widgets.dll>
 35> QApplicationPrivate::notify_helper [0x7ffe354555c0] <Qt5Widgets.dll>
 36> QApplication::notify [0x7ffe35454643] <Qt5Widgets.dll>
 37> QCoreApplication::notifyInternal2 [0x7ffe334c318a] <Qt5Core.dll>
 38> QGuiApplicationPrivate::processMouseEvent [0x7ffe339047c3] <Qt5Gui.dll>
 39> QWindowSystemInterface::sendWindowSystemEvents [0x7ffe338ef030] <Qt5Gui.dll>
 40> QEventDispatcherWin32::processEvents [0x7ffe3350bcfa] <Qt5Core.dll>
 41> qt_plugin_query_metadata [0x7ffe35af28d9] <qwindows.dll>
 42> QEventLoop::exec [0x7ffe334bf64c] <Qt5Core.dll>
 43> QCoreApplication::exec [0x7ffe334c2164] <Qt5Core.dll>
 44> TSmartPointerT<TXshSoundLevel>::~TSmartPointerT<TXshSoundLevel> [0x7ff688125219] <OpenToonz.exe>
 45> TSmartPointerT<TXshSoundTextLevel>::~TSmartPointerT<TXshSoundTextLevel> [0x7ff68854eab7] <OpenToonz.exe>
 46> TSmartPointerT<TXshSoundTextLevel>::~TSmartPointerT<TXshSoundTextLevel> [0x7ff68854e092] <OpenToonz.exe>
 47> BaseThreadInitThunk [0x7ffe86b2259d] <KERNEL32.DLL>
 48> RtlUserThreadStart [0x7ffe88d6af38] <ntdll.dll>
==== End ====

### OpenToonz Version

1.7

### Operating System

Windows

### GPU

Intel

### Graphics Tablet

None

**Comment by RodneyBaker** [comment]
It's fairly trivial for most users to draw rectangles and triangles without crashes so we need to find out why your system is experiencing this instability.  You aren't the only one so it'd be good to find out what the underlying problem is.

Your initial report has some tell tale signs that codec and antivirus issues might be involved in creating some incompatibility.
This is further supported by mentions of audio and TSmartPointer.

The steps that cause your Opentoonz to crash certainly don't here.
All of this suggests updating Opentoonz and checking for incompatiblities with codecs and antivirus programs will be well worth the time.

You might also consider joining one of the Opentoonz venues such as Opentoonz discord where others can help troubleshoot in real time.

Barring that what I personally would do:
- Update Opentoonz (but please see #5351 and other issues related to current builds/releases)
- Clear the Opentoonz cache (File > Clear Cache Folder (you can do this before updating Opentoonz but the update is more likely to cure pesky problems)
- Check antivirus settings to make sure those programs (Windows Defendener if no other antivirus programs are running) aren't being overly zealous in protecting the system.
- Investigate any codec packages that might have introduced troublesome codecs.  The most often encountered with Opentoonz seems to be related to Windows AVI codec.  If you have the K-Lite Codec pack installed... that's a prime suspect.

**Comment by WumiHam** [comment]
Rodney, thanks for your advices. I've tried them all (including nightly build of november 25, stop windows 11 defender), but the result remains still the same. Also my suspicion, that Microsoft One Drive might be the cause for crashes, didn't verify.
Opentoonz works fine as long as I do not use vector levels. I can load new vector levels and store the .pli files. And it seems, that the crash does not appear with every kind of vectors. No problems with drawing lines and moving and turning them around. But as soon as I draw filled rectangles, save them, and try to move them around, I get the crash with the meanwhile well known backtrack. When I then restart Opentoonz and load the crashed scene, I get this message:      
![Screenshot 2025-01-03 092705](https://github.com/user-attachments/assets/e7ab6c48-30af-4d52-aaf2-088239353813)

**Comment by RodneyBaker** [comment]
Yes, virtual drives such as One Drive are not compatible with serving files to Opentoonz as well as many other programs.
We'll have to investigate if anything has changed to improve that situation.

**Comment by RodneyBaker** [comment]
More importantly than the potential issue with One Drive is the fact that your screenshot is showing that your scene does not belong to a project.  This can be see by the empty Brackets in the scene/project listing at the top left of Opentoonz.  

Try going to Project Management and specify a project or create a new project and import your current scene into that new project.

There were/are some releases of Opentoonz that had issues with correctly creating and maintaining project locations.

This seems to me to be your current problem.

**Comment by WumiHam** [comment]
Looks for me like a different problem than the above mentioned crashing with vectors.  
1. Start OT
2. Set project to <project1>
3. Load scene: Scene [project1]: scene loaded correctly
4. Load a scene of <project2>  using Change Project option: scene loaded but OT looses project path (empty brackets)
5. From now on all loading of scenes results in loosing the project path. 
![Screenshot 2025-01-05 114843](https://github.com/user-attachments/assets/f90a3beb-bfc8-41ce-863a-042f94f66741)

6. To continue I have to restart OT by dubble click on a scene in Windows Explorer.     

The "vector-crash" occures when project path is correctly assigned.

**Comment by RodneyBaker** [comment]
Yes, doesn't look related to vectors although perhaps there are some additional problems you are running into there as well.

I believe the only fix is to install a release that does not have this problem.

Or rather than install, give the current portable as that shouldn't have the problem.
Link:  https://github.com/opentoonz/opentoonz/releases/download/untagged-b738019c46eb521eca78/OTportable.zip

No need to install, just download, unzip and run the Opentoonz.exe inside the directory.
Note that projects will get saved into the portablestuff directory instead of installed C:\Opentoon stuff.

**Comment by RodneyBaker** [comment]
Just to be clear. The following steps don't lead to any crash for me:

> New layer - drawing a black rectangle - expand to frame 13
> New layer - drawing a nearly white triangle on top of rectangle
> Expand triangle to frame 13
> after a few seconds: program crashed.

It would indeed be very strange if it did result in a crash.
Very specific steps there so I'm not sure about all the variables.
If you expand to only frame 12 does it still crash?
If the rectangles are different colors... still crash?

You don't mention using the Geometry Tool but I fully expect you are using that tool.
There were some issues with Geometry Tool in the past (and its unlikely all potential problems have been fixed).   I suspect you might just be using a release that had a problem that... hopefully... is now fixed.  I guess, drawing the triangle would be my place of focus if I were encountering the problem.
Not sure.

**Comment by WumiHam** [comment]
Yes, I use the Geometric Tool and it's really strange. I've tried a lot of silly things with geometric forms expanding, moving, scaling and turning around, different colors, rectangle, polygons of 3,4,5, changing its shape using the control point editor and so on. But  I cannot figure out any rule that tells me when it comes to a crash - it suddenly does - and when not.
Thus I will work around and try again with later versions. I already have moved from the latest official version to a nightly build of  November 25. I will wait a while and then trying a later version. 
The issue that bothers me more in the moment, is the loss of project path assignment when loading a scene from a project different to the current one.. But there is a work-around as well.
Thank you so far.

**Comment by RodneyBaker** [comment]
There are a few users that recommend moving to a nightly from April timeframe as that appears to be before some build issues and bugs were introduced.  You may still need to rename the SRV directory with those nightlies but its nice that moving to another nightly is an option for cases such as where persistently encountered issues aren't being resolved.

**Comment by Abel032** [comment]
> The issue that bothers me more in the moment, is the loss of project path assignment when loading a scene from a project different to the current one.. But there is a work-around as well.

This might caused by commit 6034e227 , but should be repaired by commit 2333a762

**Comment by RodneyBaker** [comment]
@WumiHam Have you been able to move on from this problem.
Specifically the issues with project paths.

**Comment by WumiHam** [comment]
I can't say because I didn't work on OT for a while. But I will now continue and try a the latest nightly.

**Comment by WumiHam** [comment]
Nightly from Mar 14 loaded:
- no more issue with project paths 
- but still crashes moving vectors around and I have no ideas what might be the reason
